system/jlib/jstream.cpp

```diff
@@ -935,6 +936,95 @@ IBufferedSerialOutputStream * createBufferedOutputStream(ISerialOutputStream * o
     return new CBlockedSerialOutputStream(output, blockWriteSize);
 }
 
+// Class that implements IBufferedSerialOutputStream with CRC functionality
+class CCrcBlockedSerialOutputStream final : public CInterfaceOf<IBufferedSerialOutputStream>
```

Comment:
I know CBlockedSerialOutputStream is a CInterfaceOf, but unless you need beforeDispose, then CSimplerInterfaceOf is preferred.